### DISCUSSION TITLE
Multi line values are getting converted into single line

### DISCUSSION THREAD

**Discussion by Bennet-Sunder** [mikefarah/yq#1584] [Q&A] [answered]
Suppose I have two files like so 

main.yml

```
---
  added: asdfasdf

```

updated_main.yml

```
---
  added:
    from: This is a huge line across
        two files
    now: |
        This is a key across two
        lines but it's with a multi line comment
    now_again: >
        This is a key across two
        lines but it's with a multi line comment
```

I wish to merge these into another file, so I use this command. 

`yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)'main.yaml updated_main.yaml > "new_file.yaml"`

And this is how `new_file.yml` looks like 

```
---
added:
  from: This is a huge line across two files
  now: |
    This is a key across two
    lines but it's with a multi line comment
  now_again: >
    This is a key across two lines but it's with a multi line comment


```

As you can see the `from, now_again` key is collapsed from two lines to one line, unlike the `now` key. I'm guessing since the `now` has the `|` symbol indicating that it is a multi-line value. 

Do we have some option that we can use as `yq` not to collapse strings like `from, now_again` into a single line? Since we have huge files and the diff gets out of hand and fixing them is overhead as well. 

Any solutions or workarounds, pls?

**Comment by mikefarah** [comment, marked as answer]
This is a known (and annoying) issue with the underlying go-yaml parser. I have seen some people do some tricky magic using diff here: https://github.com/mikefarah/yq/issues/515 can't comment on how well it works though

**Reply by Bennet-Sunder** [reply]
thanks for the reply @mikefarah 
So in case it collapses multi-line it into a single line, is it possible to at least set the line length to 80 characters so that it doesn't end up as on long string.

**Reply by mikefarah** [reply]
Nope :( There a whole bunch of issues raised against go-yaml (e.g. https://github.com/go-yaml/yaml/issues/827) but afaik none have yet been resolved :(

**Reply by mikefarah** [reply]
At some point I need to look at moving away from the lib - but it will be a big task

**Reply by Bennet-Sunder** [reply]
okaay gotcha. thanks for your time :)

**Reply by AvallaSD** [reply]
> At some point I need to look at moving away from the lib - but it will be a big task

Hi! Any updates on this?)
